pkg/apis/datasource/v0alpha1/types.go

```diff
@@ -0,0 +1,58 @@
+package v0alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/grafana/grafana/pkg/apis"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+)
+
+const (
+	GROUP   = "*.datasource.grafana.app"
+	VERSION = "v0alpha1"
+)
+
+var GenericConnectionResourceInfo = apis.NewResourceInfo(GROUP, VERSION,
+	"connections", "connection", "DataSourceConnection",
+	func() runtime.Object { return &DataSourceConnection{} },
+	func() runtime.Object { return &DataSourceConnectionList{} },
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type DataSourceConnection struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The display name
+	Title string `json:"title"`
+
+	// Optional description for the data source (does not exist yet)
+	Description string `json:"description,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type DataSourceConnectionList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []DataSourceConnection `json:"items,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type HealthCheckResult struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// The string description
+	Status string `json:"status,omitempty"`
+
+	// Explicit status code
+	Code int `json:"code,omitempty"`
+
+	// Optional description for the data source (does not exist yet)
+	Message string `json:"message,omitempty"`
+
+	// Depends on the explicit kind
+	Details *simplejson.Json `json:"details,omitempty"`
```

Comment:
same question here re: `simplejson.Json`

Reply:
I replaced our simplejson use with https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured -- stil has openapi issues, but at least standard, and the helpers are even a bit more robust